src/renderer/views/wallet/PoolShareView.tsx

```diff
@@ -40,23 +40,27 @@ export const PoolShareView: React.FC = (): JSX.Element => {
 
   const thorchainContext = useThorchainContext()
   const oRuneAddress = useObservableState(thorchainContext.address$, O.none)
-  const bitcoinContext = useBitcoinContext()
-  const oBitcoinAddress = useObservableState(bitcoinContext.address$, O.none)
-  const binanceContext = useBinanceContext()
-  const oBinanceAddress = useObservableState(binanceContext.address$, O.none)
-  /**
-   * We have to get a new stake-stream for every new asset
-   * @description /src/renderer/services/midgard/shares.ts
-   */
-  const poolShares$: PoolSharesLD = useMemo(
+  const { addressByChain$ } = useChainContext()
+
+  const [poolSharesRD]: [PoolSharesRD, unknown] = useObservableState(
     () =>
       FP.pipe(
-        sequenceTOption(oBitcoinAddress, oBinanceAddress),
-        O.fold(() => Rx.EMPTY, combineSharesByAddresses$)
+        ENABLED_CHAINS,
+        A.map(addressByChain$),
+        (addresses) => Rx.combineLatest(addresses),
+        RxOp.switchMap(
+          FP.flow(
+            A.filterMap(FP.identity),
```

Comment:
Why `A.filterMap(FP.identity)` ^ ?

Reply:
previously we had 
```ts
 sequenceTOption(oBitcoinAddress, oBinanceAddress),
 O.fold(() => Rx.EMPTY, combineSharesByAddresses$)
```
but during development not every chain address is `O.some('stringAddress')` but there might me `O.none` which would fail all sequence of `sequenceTOption(oBitcoinAddress, oBinanceAddress...),` which is unacceptable. with `filterMap(FP.identity)` we filter up `O.none` values and unwrap values to the plain `string[]` at a single place